File: coordonnees.cpp
#include "coordonnees.h"

Coordonnees::Coordonnees(int m_x, int m_y){
    x= m_x;
    y=m_y;
}
void Coordonnees::setX(int m){
    x= m;
}
void Coordonnees::setY(int m){
    y=m;
}
int Coordonnees::getX() const{
    return x;
}
int Coordonnees::getY() const{
    return y;
}

Coordonnees& Coordonnees:: operator=(const Coordonnees coord){
    this->x= coord.getX();
    this->y= coord.getY();
    return *this;
}


Coordonnees:: Coordonnees(const Coordonnees& coord,  string d){
    if (d=="haut"){
        this->x = coord.x - 1;
        this->y = coord.y;
    }
    if (d=="bas"){
        this->x = coord.x + 1;
        this->y = coord.y;
    }
    if (d=="gauche"){
        this->x = coord.x;
        this->y = coord.y - 1;
    }
    if (d=="droite"){
        this->x = coord.x;
        this->y = coord.y + 1;
    }
}

File: Coordonnees.java
package code;

import javax.swing.JPanel;

public class Coordonnees {

	//Attributs
		private	int i=0;
		private	int j=0;
		public 	MonPanel P;
		/* Constructeurs par défault
		 */
		public Coordonnees(){}
		/* Constructeurs 
		 * @param x,y : entiers correspondants
		 */
		public Coordonnees(int x,int y){
			i=x;
			j=y;
		}
		/*Constructeur après déplacement*/
		public Coordonnees(Coordonnees coord,String d  ){
			if (d=="haut"){
		        this.i = coord.i - 1;
		        this.j = coord.j;
		    }
		    if (d=="bas"){
		    	this.i = coord.i + 1;
		    	this.j = coord.j;
		    }
		    if (d=="gauche"){
		    	this.i = coord.i;
		    	this.j = coord.j - 1;
		    }
		    if (d=="droite"){
		    	this.i = coord.i;
		    	this.j = coord.j + 1;
		    }
		}
		// Accesseurs
		public int getI(){return i;}
		public int getJ(){return j;}

		// Mutateurs
		public void setI(int i){this.i=i;}
		public void setJ(int j){this.j=j;}

		// Redefinition de la methode = 
		
	 	
	 	// Redefinition de la methode hashCode
		@Override
	 	public int hashCode() {
	 		return i*10+j;
	 	}
}
